components/drivers/misc/adc.c

```diff
@@ -156,17 +159,20 @@ rt_err_t rt_adc_disable(rt_adc_device_t dev, rt_uint32_t channel)
 
 rt_uint32_t rt_adc_voltage(rt_adc_device_t dev, rt_uint32_t channel)
 {
-   rt_uint32_t value = 0, voltage = 0;
+    rt_uint32_t value = 0, voltage = 0, vref = 0;
 
     RT_ASSERT(dev);
 
     /*read the value and convert to voltage*/
     if (dev->ops->get_resolution != RT_NULL && dev->ops->convert != RT_NULL)
     {
+        /*get reference voltage*/
+        vref = _adc_control((rt_device_t) dev, RT_ADC_CMD_GET_VREF, RT_NULL );
+
         /*get the convert bits*/
         rt_uint8_t resolution = dev->ops->get_resolution(dev);
         dev->ops->convert(dev, channel, &value);
-        voltage = value * REFER_VOLTAGE / (1 << resolution);
```

Comment:
Need to change:
1. use _adc_control to instead dev->ops->get_resolution and dev->ops->convert, which can give you a clear error code.
2. need to fill in the `arg` parameter